web/modules/custom/dpl_event/dpl_event.module

```diff
@@ -287,3 +290,50 @@ function dpl_event_gin_content_form_routes() : array {
     'entity.eventinstance.edit_form',
   ];
 }
+
+/**
+ * Implements hook_entity_query_TAG_alter().
+ *
+ * Alter vocabulary queries to hide the vocabulary when the screens feature
+ * is disabled.
+ */
+function dpl_event_entity_query_taxonomy_vocabulary_alter(QueryInterface $query): void {
+  if (!\Drupal::config(SettingsForm::CONFIG_NAME)->get('enable_screen_name')) {
+    $query->condition('vid', 'screen_name', '<>');
+  }
+}
```

Comment:
Consider elaborating on how you ended up with this solution.

To be this approach came up as a bit of a surprise. I would have expected it to be handed by altering the access control system but I guess this will work too. I do not see us using hook query alter much.

I would have expected this to be mentioned in the commit comment or in an inline comment.

Reply:
> To be this approach came up as a bit of a surprise. I would have expected it to be handed by altering the access control system but I guess this will work too.

It *is* altering the access control system. The vocabularies page use a `EntityListBuilder` which relies on the entity access system (the famous `->accessCheck(TRUE)`, which uses query tags.

The documentation leaves something to be desired, but after digging around and reading code I think I ended up at https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Entity%21entity.api.php/function/hook_entity_access/11.x which points towards `hook_query_TAG_alter()`... Which is the way to actually affect the queries.

I'll expand the comment a bit.